src/annotator/components/test/NotebookModal-test.js

```diff
@@ -0,0 +1,107 @@
+import NotebookModal from '../NotebookModal';
+import { EventBus } from '../../util/emitter';
+import { mount } from 'enzyme';
+import { act } from 'preact/test-utils';
+
+describe('NotebookModal', () => {
+  let container;
+  let components;
+  let eventBus;
+
+  const createComponent = config => {
+    const component = mount(
+      <NotebookModal
+        eventBus={eventBus}
+        config={{ notebookAppUrl: '/notebook', ...config }}
+      />,
+      {
+        attachTo: container,
+      }
+    );
+    components.push(component);
+    return component;
+  };
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    components = [];
+    document.body.appendChild(container);
+    eventBus = new EventBus();
+  });
+
+  afterEach(() => {
+    components.forEach(component => component.unmount());
+    container.remove();
+  });
+
+  it('hides modal on first render', () => {
+    const wrapper = createComponent();
+    const outer = wrapper.find('.Notebook__outer');
+
+    assert.isTrue(outer.hasClass('is-hidden'));
+    assert.equal(getComputedStyle(wrapper.getDOMNode()).display, 'none');
+
+    // It also contains the iframe, although doesn't have a group.
```

Comment:
I think we should avoid rendering the iframe until the user explicitly opens the notebook. There is a resource cost to it (in terms of memory usage, XHR requests etc.) and I don't think it makes sense to incur those until the user is actually wants to use the notebook. Also I think when debugging the network requests coming from the notebook are going to be confusing if the dev is not actually testing the notebook.

Reply:
I agree, thanks. I have addressed this on a second PR.